website/views/shared/tasks/task_view/index.jade

```diff
@@ -37,7 +37,8 @@
     label(for='box-{{::obj._id}}_{{::task._id}}')
 
 // main content
-.task-text(ng-dblclick='task._editing ? saveTask(task) : editTask(task)')
+// don't allow a double click to edit if the object is locked
+.task-text(ng-dblclick='obj._locked ? {} : task._editing ? saveTask(task) : editTask(task, obj)')
```

Comment:
For challenges obj is defined here: https://github.com/HabitRPG/habitrpg/blob/develop/website/views/options/social/challenges.jade#L54

For users, obj is defined here:
https://github.com/HabitRPG/habitrpg/blob/develop/website/views/main/index.jade#L4

So I'd create a function in the challenges controller and the tasks controller (it's okay that we have duplication for it, we're about to do a rewrite to the frontend code which will make these much more component based).

```js
$scope.doubleClickTask = function (obj, task) {
  if (obj._locked) {
    return false;
  }

  if (task._editing) {
    $scope.saveTask(task);
  } else {
    $scope.editTask(task);
  }
}
```

Then in the view:

```jade
.task-text(ng-dblclick='doubleClickTask(obj, task)')
```